openedx_user_groups/views.py

```diff
@@ -0,0 +1,67 @@
+"""REST API to manage user groups, their criteria and membership."""
+
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.mixins import CreateModelMixin, ListModelMixin, RetrieveModelMixin
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.viewsets import GenericViewSet, ModelViewSet
+
+from openedx_user_groups.api import (
+    create_group_with_criteria,
+    create_group_with_criteria_and_evaluate_membership,
+    get_available_registered_criteria_schema,
+    get_group_by_id,
+)
+from openedx_user_groups.models import UserGroup
+from openedx_user_groups.serializers import UserGroupSerializer
+
+
+class AvailableCriteriaView(APIView):
+    """View to get all available criteria with their schema for fields, operators and descriptions."""
+
+    def get(self, request):
+        """Get all available criteria with their schema for fields, operators and descriptions."""
+        criteria_data = get_available_registered_criteria_schema()
+        return Response(criteria_data, status=status.HTTP_200_OK)
+
+
+class UserGroupViewSet(
```

Comment:
We're going to need to permission check this based on the scopes the user has permissions to operate on, right?

Reply:
Correct!